engine/runtime/src/main/java/org/enso/interpreter/epb/node/ForeignEvalNode.java

```diff
@@ -130,9 +127,10 @@ private void parsePy(ContextReference<EpbContext> ctxRef) {
           code.getForeignSource().lines().map(l -> "    " + l).collect(Collectors.joining("\n"));
       Source source =
           Source.newBuilder(code.getLanguage().getTruffleId(), head + indentLines, "").build();
-      CallTarget ct = ctxRef.get().getEnv().parsePublic(source);
+      EpbContext context = EpbContext.get(this);
+      CallTarget ct = context.getEnv().parsePublic(source);
       ct.call();
-      Object fn = ctxRef.get().getEnv().importSymbol("polyglot_enso_python_eval");
+      Object fn = context.getEnv().importSymbol("polyglot_enso_python_eval");
```

Comment:
This change, on the other hand, is fine: there's no context-switching here. It's written this way, because in the past Python was single-threaded and then it wasn't, but whoever was migrating this code didn't know the subtlety I'm describing and left it unchanged.